packages/explat-client/src/internal/requests.ts

```diff
@@ -72,7 +72,7 @@ export async function fetchExperimentAssignment(
 	}
 
 	if ( fetchedExperimentAssignments.length === 0 ) {
-		throw new Error( 'Received no experiment assignments while trying to fetch exactly one.' );
+		return ExperimentAssignments.createFallbackExperimentAssignment( experimentName );
```

Comment:
Since we have the `ttl` from the server at this point, we can add it to the fallback assignment as noted in https://github.com/Automattic/wp-calypso/pull/50218#issuecomment-781222347.